---
title: "Guillermo's Portfolio"
---

## About me

<img src="images/headshot.jpg" align="left" width="30%" style="float:left; padding-right:20px; padding-top:6px"/>

I am a bioinformatician with a strong interest in human health, software development and project management. Experienced in R and analysis of omics data, always eager to learn the tools and write the code needed to research and fight disease.

As a postdoctoral researcher in [Christiana Ruhrberg's lab](https://www.ucl.ac.uk/brain-sciences/ioo/research/research-labs-and-groups/ruhrberg-lab) at UCL, I analyse bulk and single-cell RNAseq data from patients with peripheral artery disease and from mouse models that reproduce this disease. This work is funded by the Leducq Foundation and by the British Heart Foundation.

In this website, you can find links to [my repositories](https://github.com/g-pinel?tab=repositories), which contain code to easily run and automate various transcriptomic analyses, source code from my Shiny apps, and others. On the [publications page](https://g-pinel.github.io/portfolio_guillermo/publications.html) you can find information about my scientific articles.

<br>

## Repositories

### E8.5 HaematoEC (web application)

<https://github.com/g-pinel/EMPs/>

This repository contains the script used to run the following Shiny App:
<br>
<https://g-pinel.shinyapps.io/Haematoendothelial_E8_5/>

This web app is a companion to our publication *Discovery of New Markers for Haemogenic Endothelium and Haematopoietic Progenitors in the Mouse Yolk Sac* ([pubmed link](https://pubmed.ncbi.nlm.nih.gov/41562864/)). (See my [publications page](https://g-pinel.github.io/portfolio_guillermo/publications.html) for further information).

It allows exploration of **single-cell RNAseq data from endothelial and haematopoietic cells from embryonic day 8.5 mouse samples** using our refined annotations.

You can query any gene to generate plots showing normalised expression levels and the absolute and relative number of cells in which the gene was detected, by cluster. A table showing subcellular location information for the proteins encoded by queried genes is also shown. In addition, you can use it to run differential expression analyses to identify markers specific to one cluster, or to compare one or more clusters against others (customizable).

<img src="images/emp_shinyapp.png" align="left" width="100%" style="float:none; padding-top:5px; padding-bottom:30px"/>

### BulkECexplorer (web application)

<https://github.com/ruhrberg/BulkECexplorer>

This repository contains the script used to run the following Shiny App:
<br>
<https://ruhrberglab.shinyapps.io/BulkECexplorer/>

This web appplicaton was published in the article *The BulkECexplorer compiles endothelial bulk transcriptomes to predict functional versus leaky transcription* ([pubmed link](https://pubmed.ncbi.nlm.nih.gov/38708406/)). (See my [publications page](https://g-pinel.github.io/portfolio_guillermo/publications.html) for further information).

With this app, you can access and analyse gene expression across the 240 samples in our endothelial cell (EC) bulk RNAseq data compendium. These samples include 5 EC subtypes commonly used in vascular biology research: HUVECs and HDMECs (human); lung, brain and retina ECs (mouse). The resulting dashboard, apart from **gene expression levels**, shows **predictions for active (functional) versus leaky (non-functional) gene expression** using three different classification methods. 

<img src="images/bulkecexplorer_app1.png" align="left" width="66%" style="float:none; padding-top:5px"/>
<img src="images/bulkecexplorer_app2.png" align="left" width="33%" style="float:none; padding-top:5px"/>

### runDESeq2

<https://github.com/g-pinel/runDESeq2>

<img src="images/deseq2.webp" align="left" width="18%" style="float:left; padding-right:20px; padding-top:6px"/>

Bulk RNA-seq experiments often include many samples belonging to multiple experimental groups and involve performing multiple comparisons between different subsets of these samples.

The functions in this repository allow you to run **one or multiple pairwise differential expression analyses** using the R DESeq2 package.<br><br>

### runEnrichmentAnalyses

<https://github.com/g-pinel/runEnrichmentAnalyses>

<img src="images/enrichment_db_logos.png" align="left" width="60%" style="float:none; padding-top:6px; padding-bottom:0px"/>

After performing differential expression analyses to obtain a list of differentially-expressed genes (DEGs), it is common to run enrichment analyses to identify groups of genes that participate in the same process or pathway and are enriched in one of the groups being compared.

The functions in this repository allow you to easily **run enrichment analyses in R against the Gene Ontology, KEGG, Wiki Pathways and Reactome databases** using DESeq2's DeseqDataSet (DDS) objects as inputs.


### autoGSEApreranked

<https://github.com/g-pinel/autoGSEApreranked>

<img src="images/gsea.png" align="left" width="200" style="float:none; padding-right:20px; padding-top:6px"/>

GSEA (Gene Set Enrichment Analysis) is a software used to determine whether a group of genes shows differences between two biological states (e.g. treatment versus control, high versus low drug dose). In this type of analysis, all genes expressed by the input samples are sorted from those most enriched in one condition to those most enriched in the other, generating a ranked list. Then, specific groups of genes are tested against this ranked list to identify those gene sets that are enriched in one of the biological states.

However, standard GSEA analyses do not support controlling for covariates like sex and age. To overcome this, you can supply GSEA with your own ranked list of genes (GSEA Preranked). Differential expression analysis packages like R's DESeq2, allow to control for covariates, and the test statistic that they output can be used as the ranking metric for the resulting genes.

The scripts in this repository allows you to easily **generate the RNK files** needed to run GSEA preranked by supplying DESeq2 objects (as RDS files) or results tables (as CSV files). You can do this in batch for multiple input files. In addition, scripts for **running one or multiple GSEA preranked analyses** with a single command is also provided. You can use the RNK files obtained with the code in this repository, or provide your own.

You can find more information about GSEA [here](https://www.gsea-msigdb.org/gsea/index.jsp).


### qPCR analyzer

<https://github.com/g-pinel/qPCR-analyzer>

<img src="images/quantstudio.jpg" align="left" width="22%" style="float:left; padding-top:6px; padding-right:20px"/>

Process your **qRT-PCR data generated using QuantStudio** with a single command. Simply provide your QuantStudio output excel files to obtain CTs, delta CTs, and levels (2<sup>-Î”Ct</sup>). 
<br><br>
Both XLS and XLSX supported, as well as combining multiple files from the same project.